warehouse/packaging/services.py

```diff
@@ -369,28 +369,26 @@ def create_service(cls, context, request):
 
 @implementer(IProjectService)
 class ProjectService:
-    def __init__(self, session, remote_addr, metrics=None, ratelimiters=None) -> None:
+    def __init__(self, session, metrics=None, ratelimiters=None) -> None:
```

Comment:
biggest refactor here. removed remote_addr on service instantiation and made it required when calling the one method that uses it.